test that get_current_task in spawn callbacks is the outer task (#236)
agronholm/anyio: docs/versionhistory.rst

- Changed the asyncio test runner to capture unhandled exceptions from asynchronous callbacks and
  unbound native tasks which are then raised after the test function (or async fixture setup or
  teardown) completes

> - Changed, the asyncio ``TaskGroup.spawn()`` now calls the new task function immediately to be
>   consistent across backends.

Reviewer: ```suggestion
- Changed the asyncio ``TaskGroup.spawn()`` method to call the target function immediately before spawning the
  task, for consistency across backends
```
